internal/pkg/agent/vault/vault_windows.go

```diff
@@ -40,16 +41,33 @@ func New(path string) (*Vault, error) {
 		path = filepath.Join(dir, path)
 	}
 
-	err := os.MkdirAll(path, 0750)
-	if err != nil {
-		return nil, err
+	if options.readonly {
+		fi, err := os.Stat(path)
+		if err != nil {
+			return nil, err
+		}
+		if !fi.IsDir() {
+			return nil, fs.ErrNotExist
+		}
+	} else {
+		err := os.MkdirAll(path, 0750)
+		if err != nil {
+			return nil, err
+		}
+		err = systemAdministratorsOnly(path, false)
+		if err != nil {
+			return nil, err
+		}
 	}
-	err = systemAdministratorsOnly(path, false)
-	if err != nil {
-		return nil, err
+
+	var entropy []byte
+
+	if options.readonly {
+		entropy, err = getSeed(path)
+	} else {
+		entropy, err = createSeedIfNotExists(path)
 	}
```

Comment:
[Suggestion]
Same as for the linux version:
Merge both if statements as both are checking the same think. Perhaps you could assign 1getSeed/createSeedIfNotExists1 to a variable if you want to call it later, outside the if block.